src/coreclr/debug/ee/debugger.cpp

```diff
@@ -10489,7 +10452,7 @@ bool Debugger::HandleIPCEvent(DebuggerIPCEvent * pEvent)
             DebuggerJitInfo * pDJI =  NULL;
             if ((pMethodDesc != NULL) && (pDMI != NULL))
             {
-                pDJI = pDMI->FindOrCreateInitAndAddJitInfo(pMethodDesc, NULL /* startAddr */);
+                pDJI = pDMI->FindOrCreateInitAndAddJitInfo(pMethodDesc, dac_cast<PCODE>(pEvent->BreakpointData.codeStartAddress));
```

Comment:
```suggestion
                pDJI = pDMI->FindOrCreateInitAndAddJitInfo(pMethodDesc, PINSTRToPCODE(dac_cast<TADDR>(pEvent->BreakpointData.codeStartAddress)));
```

PCODE is a slightly weird and error-prone type. On X86 and X64 it is exactly the same as the instruction address, but on ARM it is 31 bits of the address + a 1 bit flag that says whether the processor is operating in THUMB instruction mode. For example if the instruction address was 0x7f400000, when stored as a PCODE variable the value should be 0x7f400001. Annoyingly PCODE,   TADDR, and CORDB_ADDRESS are all type defined as integers so the compiler will happily let you cast between them, but a default cast won't correctly set and clear that extra bit flag on ARM. The call PINSTRToPCODE() handles doing the conversion correctly.